src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -149,6 +159,20 @@ public static string GetPathRoot(string path)
             return pathRoot <= 0 ? string.Empty : path.Substring(0, pathRoot);
         }
 
+        public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
+        {
+            ReadOnlySpan<char> result = new ReadOnlySpan<char>();
+
+            if (path.IsEmpty)
+                return path;
+
+            // Need to return the normalized directory separator
+            path = PathInternal.NormalizeDirectorySeparators(new string(path.ToArray())).AsReadOnlySpan();
```

Comment:
To convert Span to string, you can use string constructor that takes Span - you do not need to create a temporary array.

There is still going to unnecessary extra string allocation most of the time even after you do that. It would be nice to avoid it. The whole reason why these Span overloads are being added is to minimize allocations.

Reply:
should i write a span overload for NormalizeDirectorySeparators() ?

Reply:
Yes, it would be a good next step.